assignment.rb

```diff
@@ -0,0 +1,310 @@
+  def can_withdraw?(amount)
+    if self.balance >= amount
+      return true
+   else
+      return false
+   end
+  end
+
+  # Update the balance
+  def deduct(amount)
+    self.balance -= amount
+  end
+
+end
+
+
+
+# Account information would probably get loaded from a database in a more "real world"
+# application, but for now my database will consist of a array of user classes. I'll set that up here.
+andrew = User.new
+andrew.name = "andrew"
+andrew.pin = "1234"
+andrew.balance = 1000
+
+bob = User.new
+bob.name = "bob"
+bob.pin = "4321"
+bob.balance = 50
+
+charles = User.new
+charles.name = "charles"
+charles.pin = "9876"
+charles.balance = 50000
+
+account_holders = [ andrew, bob, charles ]
+
+
+
+# The machine class provides the information on the actual ATM
+class Machine
+  attr_accessor :balance
+
+  # Verify that the ATM contains sufficient money for the withdrawal
+  def can_withdraw?(amount)
+    if self.balance >= amount
+      return true
+   else
+      return false
+   end
+  end
+
+  # Update the balance 
+  def deduct(amount)
+    self.balance -= amount
+  end
+
+end
+
+# OK, lets initialize this ATM with a $10,000 balance
+atm = Machine.new
+atm.balance = 10000
+
+
+
+# The program class provides the actual functionality of the program.
+class Program 
+
+  attr_accessor :exit_prog
+  attr_accessor :is_valid
+  attr_accessor :usr_name
+  attr_accessor :usr_pin
+  attr_accessor :main_menu_sel
+  attr_accessor :acc_menu_sel
+
+  def initialize(machine, account_holders)
+    @atm = machine
+    @account_holders = account_holders
+    self.exit_prog = false
+    self.is_valid = false
+    self.main_menu_sel = 0
+    self.acc_menu_sel = 0
+  end
+
+  # This is the main loop that is called to run the program.
+  def run
+    # Reset defaults to prevent odditties
+    self.is_valid = false
+    self.main_menu_sel = 0
+    self.acc_menu_sel = 0
+
+    # Display the main menu and get the users input
+    main_menu_sel = self.main_menu
+
+    # Verify the main_menu_sel is a valid input, proceed only if able.
+    if (1..3).include?(main_menu_sel)
+      main_men_sel_valid = true
+      while main_men_sel_valid == true
+        case main_menu_sel
+        when 1
+          # First, lets get the user's login information
+          # This will return a hash with the user's name and pin
+          display_acc_menu = false
+          login = self.get_login_info
+
+          # Next let's try to verify that information and determine our course of action.
+          index = 0
+          ver_user = 0
+          @account_holders.each do |user|
+            if user.name == login[:name] && user.pin == login[:pin]
+              puts "Your account has been verified."
+              puts ""
+              display_acc_menu = true
+              ver_user = index
+            end
+            index += 1
+          end
+
+          # For easier access later on, let's identify which user we have verified.
+          user = @account_holders[ver_user]
```

Comment:
What advantage do you gain by setting an index on line `126` and not just setting a `current_user = user` there, instead?